app/models/user_role.rb

```diff
@@ -58,7 +58,7 @@ def cache_user_roles!
   end
 
   def build_user_role_cache
-    [ self.cached_user_roles.build(:user => owner, :role => role) ]
+    [ self.cached_user_roles.build(:user_id => owner.id, :role_id => role.id) ]
```

Comment:
What's this for exactly?  Is it to workaround an issue such as the one in #2790?